src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -54,6 +54,13 @@ void symbol_factoryt::gen_nondet_init(
     const pointer_typet &pointer_type=to_pointer_type(type);
     const typet &subtype = pointer_type.subtype();
 
+    if(subtype.id() == ID_code)
+    {
+      assignments.add(
```

Comment:
:pick: I'd still like a comment to the effect 
```
// leave as nondet_ptr to allow `remove_function_pointers`
// to replace the pointer. 
```

Reply:
Added.